packages/core/content-manager/admin/src/history/components/VersionInputRenderer.tsx

```diff
@@ -323,13 +332,23 @@ const VersionInputRenderer = ({
     case 'blocks':
       return <BlocksInput {...props} hint={hint} type={props.type} disabled={fieldIsDisabled} />;
     case 'component':
+      const { layout } = componentsLayout[props.attribute.component];
+      // Components can only have one panel, so only save the first layout item
+      const [remainingFieldsLayout] = getRemaingFieldsLayout({
+        layout: [layout],
+        metadatas: configuration.components[props.attribute.component].metadatas,
+        fieldSizes,
+        schemaAttributes: components[props.attribute.component].attributes,
+      });
+
       return (
-        <ComponentInput
-          {...props}
-          hint={hint}
-          disabled={fieldIsDisabled}
-          renderInput={(props) => <VersionInputRenderer {...props} shouldIgnoreRBAC={true} />}
-        />
+        <ComponentInput {...props} hint={hint} disabled={fieldIsDisabled}>
+          {(name) => (
+            <ComponentLayout name={name} layout={[...layout, ...remainingFieldsLayout]}>
```

Comment:
Why does ComponentLayout need to be a function receiving name instead of a component receiving name? Isn't it receiving name via the props passed to ComponentInput? Couldn't you just pass it directly here: `<ComponentLayout name={props.name} ... />`

Reply:
Do we need to re-export the Component layout? it's basically just a grid and instead of trying to be DRY it might be a lot clearer to just re-do the rendering logic here with the raw grid components, then you're only concerened with the inputs?